tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/EtcBuilder.scala

```diff
@@ -0,0 +1,98 @@
+package at.forsyte.apalache.tla.typecheck.etc
+
+import at.forsyte.apalache.tla.lir.UID
+import at.forsyte.apalache.tla.typecheck._
+
+/**
+  * A builder trait to conveniently construct instances STCExpr.
+  * Mix this trait to your class to construct STC expressions without pain.
+  * This class shields the user from the weird syntax of case classes that have two kinds of fields:
+  * the fields counted in equals, and the fields that are ignored in equals.
```

Comment:
That is the Java/Scala method that checks two objects for equality. Objets of case classes are tested for structural equality by default, in contrast to normal classes. However, a declaration like `case Foo(x: Int)(y: Int)` excludes `y` from the default equality test.

Reply:
Ok, thanks! So this would be obvious to an experienced Scala dev?

For my part, something that makes it clear "equals" here is referring to a Scala method and not equality checks on lambda terms would be helpful, but maybe I'll feel differently once I have some more Scala under my belt :)